mmv1/third_party/terraform/services/alloydb/resource_alloydb_secondary_cluster_test.go

```diff
@@ -0,0 +1,834 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAlloydbCluster_secondaryClusterMandatoryFields(context),
+			},
+			{
+				ResourceName:      "google_alloydb_cluster.secondary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// Invalid input check - can not add automated backup policy to a secondary cluster
+				Config:      testAccAlloydbCluster_secondaryClusterAddAutomatedBackupPolicy(context),
+				ExpectError: regexp.MustCompile("cannot enable automated backups on secondary cluster until it is promoted"),
+			},
+		},
+	})
+}
+
+func testAccAlloydbCluster_secondaryClusterAddAutomatedBackupPolicy(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_alloydb_cluster" "primary" {
+  cluster_id = "tf-test-alloydb-primary-cluster%{random_suffix}"
+  location   = "us-central1"
+  network    = google_compute_network.default.id
+  cluster_type = "PRIMARY"
+}
+
+resource "google_alloydb_instance" "primary" {
+  cluster       = google_alloydb_cluster.primary.name
+  instance_id   = "tf-test-alloydb-primary-instance%{random_suffix}"
+  instance_type = "PRIMARY"
+
+  machine_config {
+    cpu_count = 2
+  }
+
+  depends_on = [google_service_networking_connection.vpc_connection]
+}
+
+resource "google_alloydb_cluster" "secondary" {
+  cluster_id   = "tf-test-alloydb-secondary-cluster%{random_suffix}"
+  location     = "us-east1"
+  network      = google_compute_network.default.id
+  cluster_type = "SECONDARY"
+
+  continuous_backup_config {
+    enabled = false
+  }
+
+  secondary_config {
+    primary_cluster_name = google_alloydb_cluster.primary.name
+  }
+  
+  automated_backup_policy {
+    location      = "us-central1"
+    backup_window = "1800s"
+    enabled       = true
+
+    weekly_schedule {
+      days_of_week = ["MONDAY"]
+
+      start_times {
+        hours   = %{hour}
+        minutes = 0
+        seconds = 0
+        nanos   = 0
+      }
+    }
+
+    quantity_based_retention {
+      count = 1
+    }
+
+    labels = {
+      test = "tf-test-alloydb-secondary-cluster%{random_suffix}"
+    }
+  }
+
+  depends_on = [google_alloydb_instance.primary]
+}
+
+data "google_project" "project" {}
+
+resource "google_compute_network" "default" {
+  name = "tf-test-alloydb-secondary-cluster%{random_suffix}"
+}
+
+resource "google_compute_global_address" "private_ip_alloc" {
+  name          =  "tf-test-alloydb-secondary-cluster%{random_suffix}"
+  address_type  = "INTERNAL"
+  purpose       = "VPC_PEERING"
+  prefix_length = 16
+  network       = google_compute_network.default.id
+}
+
+resource "google_service_networking_connection" "vpc_connection" {
+  network                 = google_compute_network.default.id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.private_ip_alloc.name]
+}
+`, context)
+}
+
+func TestAccAlloydbCluster_secondaryClusterUsingCMEK(t *testing.T) {
```

Comment:
We could consolidate the other secondary settings into a `_full` test if you wanted! Setting the fields at least once is the target.

Other than `TestAccAlloydbCluster_secondaryClusterAddAutomatedBackupPolicy` since that's an error. When we get to promotion we'll want to test promoting + setting that at the same time.

Reply:
Other fields like `initial_user`, `continuous_backup_config`, `automated_backup_policy` can not be enabled or set, so keeping only basic secondary cluster.

Please let me know if you meant something else by `_full` test

Yes, will add all possible tests that exists in primary cluster to the promoted cluster tests as well

Reply:
non-blocking: We should document those fields not being available for secondary clusters somewhere.